static/app/components/events/interfaces/spans/spanGroupBar.tsx

```diff
@@ -142,7 +144,45 @@ function renderMeasurements(
 
 export function SpanGroupBar(props: Props) {
   const spanTitleRef: LegacyRef<HTMLDivElement> | null = useRef(null);
-  const {onWheel} = props;
+
+  const {
+    onWheel,
+    generateContentSpanBarRef,
+    getScrollLeftValue,
+    didAnchoredSpanMount,
+    spanGrouping,
+    toggleSpanGroup,
+  } = props;
+
+  // On mount, it is necessary to set the left styling of the content here due to the span tree being virtualized.
+  // If we rely on the scrollBarManager to set the styling, it happens too late and awkwardly applies an animation.
+
+  const setTransformCallback = useCallback(
+    ref => {
+      generateContentSpanBarRef()(ref);
+      if (ref) {
+        const left = -getScrollLeftValue();
+        ref.style.transform = `translateX(${left}px)`;
+        ref.style.transformOrigin = 'left';
+      }
+    },
+    [generateContentSpanBarRef, getScrollLeftValue]
+  );
+
+  useEffect(() => {
+    if (location.hash && !didAnchoredSpanMount) {
+      const anchoredSpanIndex = spanGrouping.findIndex(
+        span => spanTargetHash(span.span.span_id) === location.hash
+      );
+
+      // TODO: This doesn't always work.
+      // A potential fix is to just scroll to the Autogroup without expanding it if a span within it is anchored.
+      if (anchoredSpanIndex > -1) {
+        toggleSpanGroup();
+        window.scrollTo(0, window.scrollY + anchoredSpanIndex * ROW_HEIGHT);
+      }
+    }
```

Comment:
I'm going to come back to this later since this works in most cases, and I'm having trouble getting to the bottom of why it sometimes does not work.